### ISSUE TITLE
signalXY move out of view throw an exception)

### ISSUE THREAD

**Issue by goodmor9** [ScottPlot/ScottPlot#4314]
**Issue:** (signalXY move out of view throw an exception)
at this function :  GetPixelsToDrawHorizontally 
as Index out of range
 
**ScottPlot Version:5.0.39

**Code Sample:** See http://scottplot.net/faq/repro/ for tips about creating reproducible code samples

```cs
WpfPlot1.Plot.Add.SignalXY(ScottPlot.Generate.Consecutive(1000), ScottPlot.Generate.Sin(1000));
/* code to reproduce the bug */
![image](https://github.com/user-attachments/assets/2271aeac-1918-4984-acfb-9c79c5ce76ef)

**Comment by StendProg** [comment]
Hi @goodmor9,

This bug has been fixed in #4286. This fix will be included in the next release.

**Comment by goodmor9** [comment]
> Hi @goodmor9,
> 
> This bug has been fixed in #4286. This fix will be included in the next release.

can't wait.
